GDJS/Runtime/howler-sound-manager/howler-sound-manager.ts

```diff
@@ -560,42 +562,41 @@ namespace gdjs {
     ) {
       if (this._sounds[channel]) this._sounds[channel].stop();
 
-      var sound = this.createHowlerSound(
-        soundName,
-        /* isMusic= */ false
-      ).play();
-      this._sounds[channel] = sound;
-
+      const sound = this.createHowlerSound(soundName, /* isMusic= */ false);
       sound.once('play', () => {
-        sound
-          .setLoop(loop)
-          .setVolume(volume / 100)
-          .setRate(pitch);
         if (this._paused) {
           sound.pause();
           this._pausedSounds.push(sound);
         }
       });
+      // We need to first play the sound in order to change its parameters like volume.
+      sound.play();
+      sound
+        .setLoop(loop)
+        .setVolume(volume / 100)
+        .setRate(pitch);
```

Comment:
@arthuro555 @ClementPasteau Correct me if I'm wrong, but if the sound is not loaded in memory, then:

- createHowlerSound is called (no problem). It gives you back a gdjs.HowlerSound (based on a Howl that is created).
- We then call `play` on gdjs.HowlerSound. Because it is not `isLoaded` yet, then the `this._id` is not set. Only `this._howl.once('load', () => this.play())` is called.
- We then call `setLoop`/`setVolume`/`setRate`. Problem here: because the sound is not loaded, the `this._id` is null. So the volume is not applied (and same for the pitch/loop).

Conclusion: the first time the sound is loading, we risk running a sound without applying the proper settings?

I could not notice this while testing, but this seems risky - especially for long musics that are long to load if not preloaded (if we skip these, we now have a background music not at the right volume and not looping. 😬)

Reply:
I wonder if the solution is to rework the setters (currently they don't do anything if the sound is not loaded => because of this we can have invisible bugs) so that they store in the gdjs.HowlerSound the required volume/pitch/loop state, and apply it directly if they can. And if they can't it's ok, because the required volume/pitch/loop state can be applied in `play()` (i.e: as soon as you get the id => you set the volume/pitch/loop, because the source of truth is now the gdjs.HowlerSound).

Reply:
You're right for your first comment. I haven't noticed any issue either, but it could be because we only have short sounds.
As you suggested, it could be worth creating an audio playground project, with all types and sizes of sounds & musics playing on different channels so we can experiment.
I'll try to do that next so that we feel more confident pushing a fix

Reply:
Oh right that's the reason i had put it in the once in the first place. Indeed storing the settings and applying them on play sounds like the optimal way, it would also make the JavaScript api less "buggy" (as in these potential issues wouldn't be triggerable by misusing the API either).